lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/BuiltinsProcessor.java

```diff
@@ -0,0 +1,211 @@
+    String ensoMethodName = methodName.replaceAll("([^_A-Z])([A-Z])", "$1_$2").toLowerCase();
+    try (PrintWriter out = new PrintWriter(gen.openWriter())) {
+      out.println("package " + builtinNode.getPkgName() + ";");
+      out.println();
+      for (String importPkg : necessaryImports) {
+        out.println("import " + importPkg + ";");
+      }
+      out.println("import " + ownerClazz.fullyQualifiedName() + ";");
+      out.println();
+      out.println(
+          "@BuiltinMethod(type = \""
+              + ownerClazz.getName()
+              + "\", name = \""
+              + ensoMethodName
+              + "\", description = \""
+              + description
+              + "\")");
+      out.println("public class " + builtinNode.getName() + " extends Node {");
+      out.println();
+
+      for (String line : mgen.generateMethod(methodName, ownerClazz.name)) {
+        out.println("  " + line);
+      }
+
+      out.println();
+      out.println("}");
+    }
+  }
+
+  private final List<String> necessaryImports =
+      Arrays.asList("com.oracle.truffle.api.nodes.Node", "org.enso.interpreter.dsl.BuiltinMethod");
+
+  private MethodParameter fromVariableElementToMethodParameter(VariableElement v) {
+    return new MethodParameter(v.getSimpleName().toString(), v.asType().toString());
+  }
+
+  /** Method generator encapsulates the generation of the `execute` method. */
+  private class MethodGenerator {
+    private final String returnTpe;
+    private final List<MethodParameter> params;
+    private final boolean isStatic;
+
+    public MethodGenerator(ExecutableElement method) {
+      this(
+          method.getReturnType().toString(),
+          method.getParameters().stream()
+              .map(v -> fromVariableElementToMethodParameter(v))
+              .collect(Collectors.toList()),
+          method.getModifiers().contains(Modifier.STATIC));
+    }
+
+    private MethodGenerator(String returnTpe, List<MethodParameter> params, boolean isStatic) {
+      this.returnTpe = returnTpe;
+      this.params = params;
+      this.isStatic = isStatic;
+    }
+
+    public String[] generateMethod(String name, String owner) {
+      String paramsDef;
+      String paramsApplied;
+      if (params.isEmpty()) {
+        paramsDef = "";
+        paramsApplied = "";
+      } else {
+        paramsDef =
+            ", "
+                + StringUtils.join(params.stream().map(x -> x.declaredParameter()).toArray(), ", ");
+        paramsApplied =
+            ", " + StringUtils.join(params.stream().map(x -> x.getName()).toArray(), ", ");
+      }
+      String thisParamTpe = isStatic ? "Object" : owner;
+      return new String[] {
+        returnTpe + " execute(" + thisParamTpe + " _this" + paramsDef + ") {",
+        isStatic
+            ? "  return " + owner + "." + name + "(" + paramsApplied + ");"
+            : "  return _this." + name + "(" + paramsApplied + ");",
+        "}"
+      };
+    }
+  }
+
+  private class ClassName {
```

Comment:
Use `record` - now, with Frgaal, it should be available to use.

Reply:
Kinda. It's not enabled for `interpreter-dsl` project. I will make it happen once https://github.com/enso-org/enso/pull/3433 is in.

Reply:
done